qiskit/algorithms/gradients/finite_diff_estimator_gradient.py

```diff
@@ -0,0 +1,94 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Gradient of Sampler with Finite difference method."""
+
+from __future__ import annotations
+
+from typing import Sequence
+
+import numpy as np
+
+from qiskit.circuit import Parameter, QuantumCircuit
+from qiskit.opflow import PauliSumOp
+from qiskit.primitives import BaseEstimator
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+
+from .base_estimator_gradient import BaseEstimatorGradient
+from .estimator_gradient_result import EstimatorGradientResult
+
+
+class FiniteDiffEstimatorGradient(BaseEstimatorGradient):
+    """
+    Compute the gradients of the expectation values by finite difference method.
+    """
+
+    def __init__(self, estimator: BaseEstimator, epsilon: float, **run_options):
+        """
+        Args:
+            estimator: The estimator used to compute the gradients.
+            epsilon: The offset size for the finite difference gradients.
+            run_options: Backend runtime options used for circuit execution. The order of priority is:
+                run_options in ``run`` method > gradient's default run_options > primitive's default
+                setting. Higher priority setting overrides lower priority setting.
+
+        Raises:
+            ValueError: If ``epsilon`` is not float.
+        """
+        if not isinstance(epsilon, float):
+            raise ValueError(f"epsilon must be a float, but got {type(epsilon)} instead.")
```

Comment:
It may raise an error if users set an integer `1`. I think it's more useful to include the check whether `epsilion` is nonnegative or not, i.e., 
```
if epsilon <= 0:
    raise ValueError(...)
```

I don't think we need to check types of arguments unless the function behaves differently depending on the type of the argument.

Reply:
It may work with negative `epsilon` because of the symmetry. (I didn't check it)

Reply:
```
if epsilon <= 0:
    raise ValueError(f"epsilon ({epsilon}) should be non-negative")
```
This sounds nice. I added a `ValueError` just in case. Let's change  is to this. Probably, negative `epsilon` works, but the result is the same as positive one anyway.

Reply:
I see. Then, we can skip the check.